library/setup/ComposerHelper.php

```diff
@@ -33,6 +38,61 @@ private static function clearAddonManagerCache() {
         }
     }
 
+    /**
+     * Trigger builds of frontend assets after a composer install.
+     *
+     * - Installs node_modules
+     * - Builds frontend assets
+     *
+     * There are some environmental variables that can alter the way this command run.
+     *
+     * - VANILLA_BUILD_NODE_ARGS - Arguments to pass to the node process.
+     * "--max-old-space-size" in particular can be used to set a memory limit.
+     * - VANILLA_BUILD_DISABLE_CODE_VALIDATION - Disables type-checking and linting. This speeds up the build and
+     * reduces memory usage.
+     * - VANILLA_BUILD_DISABLE_AUTO_BUILD - Prevent the build from running on composer install.
+     */
+    public static function postUpdate() {
+        $skipBuild = getenv(self::DISABLE_AUTO_BUILD) ? true : false;
+        if ($skipBuild) {
+            printf("\nSkipping automatic JS build because " . self::DISABLE_AUTO_BUILD . " env variable is set.");
+            return;
+        }
+
+        printf("\nInstalling core node_modules\n");
+        passthru('yarn install --pure-lockfile', $installReturn);
+
+        if ($installReturn !== 0) {
+            printf("Installing core node_modules failed\n");
+            exit($installReturn);
+        }
+
+        $vanillaRoot = realpath(__DIR__ . "/../../");
+        $buildScript = realpath($vanillaRoot . "/build/scripts/build.ts");
+        $tsNodeRegister = realpath($vanillaRoot . "/node_modules/ts-node/register");
+        $tsConfig = realpath($vanillaRoot . "/build/tsconfig.json");
+
+
+        // Build bootstrap can be used to configure this build if env variables are not available.
+        $buildBootstrap = realpath($vanillaRoot . "/conf/build-bootstrap.php");
+        if (file_exists($buildBootstrap)) {
+            include $buildBootstrap;
+        }
+
+        $nodeArgs = getenv(self::NODE_ARGS_ENV) ?: "";
+        $disableValidationFlag = getenv(self::DISABLE_VALIDATION_ENV) ? "--low-memory" : "";
+        $buildCommand = "TS_NODE_PROJECT=$tsConfig node $nodeArgs -r $tsNodeRegister $buildScript $disableValidationFlag";
```

Comment:
I realize the chance of this happening are slim, but does this command accommodate arguments/paths with spaces in them? If not, could it?

Reply:
We're shelling out to execute this command, so you can provide a path with spaces, but you would need to escape the whitespace the way you would for any shell.

I haven't tested that scenario particularly though.